jetty-core/jetty-fcgi/jetty-fcgi-server/src/test/java/org/eclipse/jetty/fcgi/server/AbstractHttpClientServerTest.java

```diff
@@ -65,33 +58,28 @@ public void start(Handler handler) throws Exception
         QueuedThreadPool clientThreads = new QueuedThreadPool();
         clientThreads.setName("client");
         clientConnector.setExecutor(clientThreads);
-        // TODO: restore leak tracking.
         if (clientBufferPool == null)
-            clientBufferPool = new ArrayByteBufferPool();
+            clientBufferPool = new ArrayByteBufferPool.Tracking();
         clientConnector.setByteBufferPool(clientBufferPool);
         HttpClientTransport transport = new HttpClientTransportOverFCGI(clientConnector, "");
-        transport.setConnectionPoolFactory(destination -> new LeakTrackingConnectionPool(destination, client.getMaxConnectionsPerDestination())
-        {
-            @Override
-            protected void leaked(LeakDetector<Connection>.LeakInfo leakInfo)
-            {
-                connectionLeaks.incrementAndGet();
-            }
-        });
         client = new HttpClient(transport);
         client.start();
     }
 
     @AfterEach
-    public void dispose() throws Exception
+    public void dispose()
     {
-        System.gc();
-
-        assertThat("Connection Leaks", connectionLeaks.get(), Matchers.is(0L));
-
-        if (client != null)
-            client.stop();
-        if (server != null)
-            server.stop();
+        try
+        {
+            if (serverBufferPool != null)
+                assertThat("Server Leaks: " + serverBufferPool.getLeaks(), serverBufferPool.getLeaks().size(), Matchers.is(0));
```

Comment:
Use `dumpLeaks()` to better dump the leaks in case of assertion failure.

Reply:
Done.